From 9036478549729580a89db90835ee5c688acbc282 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@freescale.com>
Date: Fri, 30 Oct 2015 17:31:31 +0800
Subject: [PATCH] MLK-11800 dts: imx7d-sdb: correct flexcan stby gpio pin

According to schematic, the pad used for CAN STBY signal
should be EPDC_DATA14, not GPIO1_IO07. So correct it.
And due to pin is conflict with epdc, so we also update
the imx7d-sdb-epdc.dts.

Reported-by: Peng Fan <Peng.Fan@freescale.com>
Signed-off-by: Dong Aisheng <aisheng.dong@freescale.com>
---
 arch/arm/boot/dts/imx7d-sdb-epdc.dts | 4 ++++
 arch/arm/boot/dts/imx7d-sdb.dts      | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb-epdc.dts b/arch/arm/boot/dts/imx7d-sdb-epdc.dts
index 9dcb9d7..fc2c656 100644
--- a/arch/arm/boot/dts/imx7d-sdb-epdc.dts
+++ b/arch/arm/boot/dts/imx7d-sdb-epdc.dts
@@ -20,6 +20,10 @@
 	status = "disabled";
 };
 
+&flexcan2 {
+	status = "disabled";
+};
+
 &max17135 {
         status = "okay";
 };
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 696f9ba..b286a18 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -63,7 +63,7 @@
 			regulator-name = "can2-3v3";
 			regulator-min-microvolt = <3300000>;
 			regulator-max-microvolt = <3300000>;
-			gpio = <&gpio1 7 GPIO_ACTIVE_LOW>;
+			gpio = <&gpio2 14 GPIO_ACTIVE_LOW>;
 		};
 
 		reg_vref_1v8: regulator@3 {
@@ -930,7 +930,7 @@
 		pinctrl_hog_2: hoggrp-2 {
 			fsl,pins = <
 				MX7D_PAD_GPIO1_IO05__GPIO1_IO5	0x14
-				MX7D_PAD_GPIO1_IO07__GPIO1_IO7	0x59  /* CAN_STBY */
+				MX7D_PAD_EPDC_DATA14__GPIO2_IO14 0x59  /* CAN_STBY */
 				MX7D_PAD_GPIO1_IO00__WDOD1_WDOG_B 0x74
 			>;
 		};
-- 
1.8.0

